Marlin/Marlin_main.cpp

```diff
@@ -1589,6 +1657,12 @@ static void setup_for_endstop_move() {
         current_position[X_AXIS], current_position[Y_AXIS], zPosition,
         current_position[E_AXIS]
       );
+      #ifdef REPROBE
+        if (zPosition == MIN_PROBE_PT && !digitalRead(Z_MIN_PIN)^Z_MIN_ENDSTOP_INVERTING) {
```

Comment:
I see you're trying to keep the Z probe from going more than 2mm below the bed, but as pointed out above, that's not how this works. We don't know where Z is on the first probe, and during probing we don't know if the probe failed because the probe is broken or because it never reached the bed.